docs/source/custom_resolvers.rst

```diff
@@ -245,14 +255,18 @@ It takes two parameters:
 oc.decode
 ^^^^^^^^^
 
-Strings may be converted using ``oc.decode``:
+With ``oc.decode``, strings can be converted into their corresponding data types using the OmegaConf grammar.
+This grammar recognizes typical data types like ``bool``, ``int``, ``float``, ``dict`` and ``list``,
+e.g. ``"true"``, ``"1"``, ``"1e-3"``, ``"{a: b}"``, ``"[a, b, c]"``.
+
+Note that:
 
-- Primitive values (e.g., ``"true"``, ``"1"``, ``"1e-3"``) are automatically converted to their corresponding type (bool, int, float)
-- Dictionaries and lists (e.g., ``"{a: b}"``, ``"[a, b, c]"``)  are returned as transient config nodes (DictConfig and ListConfig)
-- Interpolations (e.g., ``"${foo}"``) are automatically resolved
-- ``None`` is the only valid non-string input to ``oc.decode`` (returning ``None`` in that case)
+- When providing as input to ``oc.decode`` a string that is meant to be decoded into another string, in general
+  the input string should be quoted (since only a subset of characters are allowed by the grammar in unquoted
```

Comment:
This is a bit cumbersome, maybe:
```suggestion
- Input strings provided to `oc.decode` should be quoted because only a subset of the characters is allowed in unquoted strings
```

(make sure it formats properly).

Reply:
I also think it would be good to show decoding of a literal input string as well (not the result of another resolver).

Reply:
> This is a bit cumbersome, maybe:
> (...)

Done in 8ea1549 (with minor changes)

> I also think it would be good to show decoding of a literal input string as well (not the result of another resolver).

I'm a bit reluctant to do so because:
* It would break a bit the "flow" of the current example
* I can't think of a situation where one would want to use `oc.decode` on a literal input string

Do you have a specific suggestion in mind?

Reply:
No, just a suggestion. we can leave it like that.